libopae++/include/opaec++/pvalue.h

```diff
@@ -55,44 +108,46 @@ struct pvalue
     {
     }
 
-    template<typename U>
-    pvalue<T>& operator=(U v){
-        T value = static_cast<T>(v);
-        auto res = set_(*props_, value);
-        if (res != FPGA_OK){
-            // TODO: Print error: std::cerr << "Could not set property";
+    template<typename V = typename std::enable_if<!std::is_same<T, guid_t>::value, T>::type>
```

Comment:
Help me understand what this is doing.

Reply:
Basically, it is enabling this templated function only when `T` is not guid_t. It also assigns the type, `T`, to `V`.
It was something I was experimenting with before when I had `guid_t` a typedef of `fpga_guid`. It's probably not needed anymore and I will most likely remove it an upcoming PR.
`::value` refers to the value of the enable_if (true or false). `::type` refers to the type argument (in this case `T`) of enable_if.